### PR TITLE
Enabled code coverage

### PR DESCRIPTION
**Pull request by pinebit**
This is for https://app.shortcut.com/chainlinklabs/story/22152/set-up-test-golang-test-coverage-for-ci

The coverage reports can be found in https://app.codecov.io/gh/smartcontractkit/chainlink

NOTE: current threshold is set to 55%, but shall be increased as we improve test coverage.

### CODE REVIEW

File: tools/bin/go_core_tests [hunk 1]
```diff
@@ -24,5 +24,7 @@ if [[ $EXITCODE != 0 ]]; then
   echo "Encountered test failures."
 else
   echo "All tests passed!"
+  # uploading coverage.txt to CodeCov
+  bash <(curl -s https://codecov.io/bash)
```
Comment (se3000) [inline tools/bin/go_core_tests:28]: Can we vendor this script instead of pulling it down every time? Maybe in `tools/bin/`.

**Review by se3000 — CHANGES_REQUESTED** [review]

File: tools/bin/go_core_tests (hunk 1 above)
Comment (pinebit) [reply tools/bin/go_core_tests:28]: I was thinking about the same, but isn't it better to use the up-to-date version from codecov?

Comment (pinebit) [reply tools/bin/go_core_tests:28]: This way we should always be compatible to codecov.

Comment (makramkd) [reply tools/bin/go_core_tests:28]: Worth noting that there was a security issue with this a bit ago (see https://about.codecov.io/security-update/).

Comment (jmank88) [reply tools/bin/go_core_tests:28]: We could use the pattern we are using on some generated code where we check in a copy, but also fetch again at runtime to verify it matches.

Comment (jmank88) [reply tools/bin/go_core_tests:28]: We could use the pattern we are using on some generated code where we check in a copy, but also fetch again at runtime to verify it matches.

Comment (pinebit) [reply tools/bin/go_core_tests]: Added codecov script

**Review by jmank88 — APPROVED** [review]

**Review by se3000 — APPROVED** [review]

**Pull request merged**